chia/simulator/simulator_test_tools.py

```diff
@@ -28,20 +28,21 @@
 from chia.util.keychain import Keychain
 from chia.util.lock import Lockfile
 from chia.wallet.derive_keys import master_sk_to_wallet_sk
+from tests.util.keyring import TempKeyring
```

Comment:
We can't import from tests in this file

Reply:
The simulator tests are designed to run standalone, not with the not exposed tests library

Reply:
Not sure I understand this - this code is located in `chia/simulator/simulator_test_tools.py` which you would get by cloning chia-blockchain, in which case you would have `tests.util.keyring`. Under what scenario would it not run?

Reply:
Any packaged result for example.  My personal projects include the tests in the main package (`chia._tests` for example) but we do not (and many Python projects don't).  Anyways, the last time this was an issue with the simulator the relevant code was moved out of the tests and into the main application.

Reply:
I modified the code to avoid this problem, I think the changes now are an improvement. - resolving the conversation